File: app/src/main/java/cat/teknos/cconverter/MainActivity.java
package cat.teknos.cconverter;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void clickExchange(View v){

        Spinner SpnBaseCurrency = findViewById(R.id.SpnBaseCurrency);
        Spinner SpnQuoteCurrency = findViewById(R.id.SpnQuoteCurrency);
        EditText baseQuantity = findViewById(R.id.baseQuantity);
        TextView convertedAmount = findViewById(R.id.convertedAmount);

        String BaseCoin = SpnBaseCurrency.getSelectedItem().toString();
        String QuoteCoin = SpnQuoteCurrency.getSelectedItem().toString();

        double changeValue = Double.parseDouble(baseQuantity.getText().toString());
        double result = exchange(BaseCoin,QuoteCoin,changeValue);

        if(result > 0){
            convertedAmount.setText(String.format(Locale.US, "For %5.2f %s, you will receive %5.2f %s",changeValue,BaseCoin,result,QuoteCoin));
            baseQuantity.setText("");
        }else{
            Toast.makeText(MainActivity.this, "This option has not an exchange value", Toast.LENGTH_SHORT).show();
        }
    }

    private double exchange(String baseCurrency, String quoteCurrency, double lastPrice){

        double convertedAmount = 0;
        double USDtoEUR = 0.95;
        double MXNtoUSD = 0.06;
        double MXNtoEUR = 0.05;

        switch (baseCurrency){
            case "USD":
                if(quoteCurrency.equals("EUR"))
                    convertedAmount = lastPrice * USDtoEUR;
                if(quoteCurrency.equals("MXN"))
                    convertedAmount = lastPrice / MXNtoUSD;
                break;

            case "EUR":
                if(quoteCurrency.equals("USD"))
                    convertedAmount = lastPrice / USDtoEUR;
                if(quoteCurrency.equals("MXN"))
                    convertedAmount = lastPrice / MXNtoEUR;

                break;

            case "MXN":
                if(quoteCurrency.equals("USD"))
                    convertedAmount = lastPrice * MXNtoUSD;
                if(quoteCurrency.equals("EUR"))
                    convertedAmount = lastPrice * MXNtoEUR;
                break;
        }
        return convertedAmount;
    }
}

File: app/src/main/java/cat/teknos/cconverter/MainActivity.kt
package cat.teknos.cconverter

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.util.Locale

class MainActivity : AppCompatActivity() {

    private val exchangeRates = mapOf(
        Pair("USD", mapOf("EUR" to 0.9444, "MXN" to 17.3692)),
        Pair("EUR", mapOf("USD" to 1.0589, "MXN" to 18.3937)),
        Pair("MXN", mapOf("USD" to 0.0576, "EUR" to 0.0544))
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupClickListeners()
    }

    private fun setupClickListeners() {
        findViewById<Button>(R.id.btnCalculate).setOnClickListener {
            val spnBaseCurrency = findViewById<Spinner>(R.id.SpnBaseCurrency)
            val spnQuoteCurrency = findViewById<Spinner>(R.id.SpnQuoteCurrency)
            val baseQuantity = findViewById<EditText>(R.id.baseQuantity)
            val convertedAmount = findViewById<TextView>(R.id.convertedAmount)

            val baseCoin = spnBaseCurrency.selectedItem.toString()
            val quoteCoin = spnQuoteCurrency.selectedItem.toString()
            val changeValueText = baseQuantity.text.toString()

            if (changeValueText.isNotBlank()) {
                try {
                    val changeValue = changeValueText.toDouble()
                    val result = exchange(baseCoin, quoteCoin, changeValue)

                    if (result > 0) {
                        convertedAmount.text = String.format(
                            Locale.US,
                            "%.2f %s",
                            result,
                            quoteCoin
                        )
                    } else {
                        Toast.makeText(
                            this@MainActivity,
                            "This option does not have an exchange value",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                } catch (e: NumberFormatException) {
                    Toast.makeText(
                        this@MainActivity,
                        "Invalid input. Please enter a valid number.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            } else {
                Toast.makeText(
                    this@MainActivity,
                    "Please enter a quantity to convert.",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }

    private fun exchange(baseCurrency: String, quoteCurrency: String, lastPrice: Double): Double {
        return exchangeRates[baseCurrency]?.get(quoteCurrency)?.let {
            if (it > 0) {
                lastPrice * it
            } else {
                -1.0
            }
        } ?: -1.0
    }
}
